### PR TITLE
feat: update asset and asset endpoint profile api to GA versions

### PR DESCRIPTION
**Pull request by vilit1**
removed all discovered asset refs for now
update the api_version for assets by hand
life

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: azext_edge/edge/vendor/clients/assetmgmt/_client.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
```
Comment (digimaun) [inline azext_edge/edge/vendor/clients/assetmgmt/_client.py:8]: Not sure if a an entire separate client is necessary if there are no breaking changes between versions. api_version kwargs can be passed to override API versions on the client

File: azext_edge/edge/vendor/clients/assetmgmt/_client.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        )
+        _policies = kwargs.pop("policies", None)
+        if _policies is None:
+            _policies = [
+                policies.RequestIdPolicy(**kwargs),
+                self._config.headers_policy,
+                self._config.user_agent_policy,
+                self._config.proxy_policy,
+                policies.ContentDecodePolicy(**kwargs),
+                ARMAutoResourceProviderRegistrationPolicy(),
```
Comment (digimaun) [inline azext_edge/edge/vendor/clients/assetmgmt/_client.py:77]: Looks like you are using different version of autorest components to generate this client, compared to our other vendored clients.

**Comment by vilit1** [comment]
image

**Comment by vilit1** [comment]
also the wonderful support bundle
image

**Review by digimaun — APPROVED** [review]

File: azext_edge/edge/providers/rpsaas/adr/asset_endpoint_profiles.py [hunk 3]
```diff
@@ -63,7 +60,6 @@ def create(
         password_reference: Optional[str] = None,
         username_reference: Optional[str] = None,
         tags: Optional[Dict[str, str]] = None,
-        discovered: bool = False,  # for easy discovered debugging
```
Comment (digimaun) [inline azext_edge/edge/providers/rpsaas/adr/asset_endpoint_profiles.py:66]: How about discovered related params? https://github.com/Azure/azure-iot-ops-cli-extension/blob/feature/1.0.0/azext_edge/edge/params.py#L616

Comment (vilit1) [reply azext_edge/edge/providers/rpsaas/adr/asset_endpoint_profiles.py:66]: true

**Pull request merged**